documentation/docs/03-loading.md

```diff
@@ -58,14 +58,16 @@ type LoadOutput = {
 
 If `load` returns nothing, SvelteKit will [fall through](#routing-advanced-fallthrough-routes) to other routes until something responds, or will respond with a generic 404.
 
-> `load` only applies to components that define pages, not the components that they import.
->
-> It is important to note that `load` may run on either the server or in the client browser. Code called inside `load` blocks:
->
-> - should use the SvelteKit-provided [`fetch`](#loading-input-fetch) method for getting data in order to avoid duplicate network requests
-> - should generally run on the same domain as any upstream API servers requiring credentials
-> - should not reference `window`, `document`, or any browser-specific objects
-> - should not reference any API keys or secrets directly, which will be exposed to the client, but instead call an endpoint using any required secrets
+> `load` only applies to [pages](#routing-pages) and [layouts](#layouts), and it may run on either the server or in the client browser.
+
+Code called inside `load` blocks:
+
+- should use the SvelteKit-provided [`fetch`](#loading-input-fetch) method for getting data in order to avoid duplicate network requests
+- should generally run on the same domain as any upstream API servers requiring credentials
+- should not reference `window`, `document`, or any browser-specific objects
+- should not reference any API keys or secrets directly, which will be exposed to the client, but instead call an endpoint using any required secrets
+
+> Any state changes to a variable that happens on the server will be saved and persisted on that server as well, and all clients hitting the server-rendered page will be exposed to the already mutated state from the server. To share states exclusive to individual clients, use [`request.locals`](#hooks-handle) or [session](#loading-input-session).
```

Comment:
"state changes to a variable" should probably be "state changes to a global variable" or something more specific

the reason to use `request.locals` is to share state across hooks, request, and page on the server. And `session` allows you to share between the server and client. I think that's a bit of info that could be useful in determining which is appropriate

Reply:
I agree with the first one. The additional bit of info should probably be placed in `locals` and `session` sections respectively instead, since we've already linked them here.

Reply:
One thing that I think could be a bit confusing about the current wording in this PR is that it talks about sharing state with clients and mentions `request.locals`, but `request.locals` only stays on the server and client typically refers to the web browser

Reply:
```suggestion
> Mutating any shared state on the server will affect all clients, not just the current one.
```

We probably shouldn't talk about `request.locals` here since `load` doesn't see them, it adds confusion